includes/class-wc-stripe-webhook-handler.php

```
@@ -515,6 +515,204 @@ public function process_webhook_dispute_closed( $notification ) {
+		if ( 'inactive' === $mandate_status && $revocation_reason ) {
+			$target_order_status = OrderStatus::CANCELLED;
+		} elseif ( 'inactive' === $mandate_status ) {
+			$target_order_status = OrderStatus::ON_HOLD;
+		}
+
+		// Idempotency guard: skip if order is already in the target status (for actionable states)
+		// or if a non-actionable state webhook is redelivered.
+		if ( $target_order_status && $order->has_status( $target_order_status ) ) {
+			WC_Stripe_Logger::info(
+				sprintf( 'Order %d already has status %s, skipping mandate update.', $order->get_id(), $target_order_status )
+			);
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'process_webhook_mandate_updated'. (undefined)

(StaticAccess)